app/src/main/java/com/dragic/beggining_recyclerview_kodeco/ui/AllFragment.kt

```diff
@@ -35,26 +35,38 @@ import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import com.dragic.beggining_recyclerview_kodeco.R
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.dragic.beggining_recyclerview_kodeco.databinding.FragmentAllBinding
+import com.dragic.beggining_recyclerview_kodeco.model.CreatureStore
 
 
 class AllFragment : Fragment() {
 
-  companion object {
-    fun newInstance(): AllFragment {
-      return AllFragment()
-    }
-  }
+    private lateinit var binding: FragmentAllBinding

```

Comment:
`lateinit var` bindings are only acceptable in Activities because their View is destroyed in onDestroy, which also destroys the Activity. 

In contrast, a Fragment's View is destroyed in onDestroyView, while the Fragment itself is destroyed in onDestroy. A Fragment can destroy and recreate its View multiple times during its lifecycle. Because of this, keeping a `lateinit var` binding can result in memory leaks where you keep a reference to the old, destroyed View. 

Please refactor this according to the documentation: https://developer.android.com/topic/libraries/view-binding#fragments

Reply:
done